docs/async/async-csharp-vb.rst

```diff
@@ -0,0 +1,167 @@
+		
+		Dim b As BasicUserData = e.Source
+		
+		' _directionsClient is a made-up object that would encapsulate certain complexity,
+		' such as getting the location from the device.
+		Dim getDirectionsTask As Task(Of String) = _directionsClient.GetDrivingDirectionsToLocation(user.LastKnownLocation)
+		
+		' This independent work can be done concurrently
+		' since it doesn't rely on directions!
+		NotifyUserOfPickupAsync(b.ContactInfo)
+		
+		Dim directionsJson As String = Await getDirectionsTask
+		
+		' Using the JSON.NET library to deserialize data
+		Dim d As DirectionsData = JsonConvert.DeserializeObject(Of DirectionsData)(directionsJson)
+		
+		DrawRouteOnMap(d.Polyline)		
+	End Sub
+
+Web Service snippet:
+
+.. code-block:: vb.net
+
+	<HttpGet>
+	Public Async Function GetDirections(startPos as GeoPoint, endPos as GeoPoint) As Task(Of String)
+
+		' Suspends GetUser() to allow the caller (the web service) to accept another request,
+		' rather than blocking on this one.
+		Dim dirs As Directions = Await _locationService.GetDirectionsAsync(startPos, endPost);
+		
+		Dim json As String = JsonConvert.SerializeObject(dirs);
+		
+		Return json
+	End Function
+
+Important Info and Advice
+-------------------------
+
+Although async programming is relatively straightfoward, there are some details to keep in mind which could otherwise result in some nasty behavior.
+
+* Do yourself a favor and append "Async" to the end of every async method you write which could be consumed by another method.
+
+Yes, it's sort of hungarian notation which is widely hated, but being extra explicit is a lot better than tracking down a race condition.  Note that certain methods which aren't explicity called by your code (such as event handlers or web controller methods) may not necessarily apply.
+
+* ``await`` is what will ultimately make a method asynchronous.
+
+Failing to apply the ``await`` operator to a task will result in the async method running synchronously!  Application of ``await`` is what suspends the async method, giving back control to the method which called it.  Pay attention to compiler warnings about this.
+
+* ``async void`` should only be used for event handlers.
+
+Why?  That's the only reason they were allowed in the first place!  Async programming uses the ``Task`` and ``Task<T>`` objects, which provide flexibility in dealing with any asynchronous work that needs to be done.  Throwing that out of the window with ``async void`` doesn't follow the model very well.  Here's some specific issues:
+
+    (a) Exceptions thrown in an ``async void`` method can't be caught.
+	
+    (b) ``async void`` methods are very difficult to test.
+	
+    (c) ``async void`` methods can cause bad side effects if the caller isn't expecting them to be async.
+	
+That said, ``async void`` is perfect for event handlers where the event involves any blocking task(s).
+
+* Avoid async lambda expressions when combined with other async code
```

Comment:
Looking at this, I can read it as "avoid async lambdas", but then below you mention lambdas only in the context of LINQ (which is a good point). If this is only about LINQ, make it clearer on this line.